tests/backends/base/test_operations.py

```diff
@@ -1,8 +1,62 @@
 from django.db import NotSupportedError, connection
+from django.db.backends.base.operations import BaseDatabaseOperations
+from django.db.models import DurationField
 from django.test import SimpleTestCase, skipIfDBFeature
 
 
 class DatabaseOperationTests(SimpleTestCase):
+    def setUp(self):
+        super().setUp()
+        self.base = BaseDatabaseOperations(connection=connection)
+
+    @skipIfDBFeature('can_distinct_on_fields')
+    def test_distinct_on_fields(self):
+        msg = 'DISTINCT ON fields is not supported by this database backend'
+        with self.assertRaisesMessage(NotImplementedError, msg):
+            connection.ops.distinct_sql(['a', 'b'])
+
+    def test_end_transaction_rollback(self):
+        sql = self.base.end_transaction_sql(success=False)
+        self.assertEqual(sql, 'ROLLBACK;')
+
+    def test_no_limit_value(self):
+        msg = (
+            'subclasses of BaseDatabaseOperations may require a '
+            'no_limit_value() method'
+        )
+        with self.assertRaisesMessage(NotImplementedError, msg):
+            self.base.no_limit_value()
+
+    def test_quote_name(self):
+        msg = 'subclasses of BaseDatabaseOperations may require a quote_name() method'
+        with self.assertRaisesMessage(NotImplementedError, msg):
+            self.base.quote_name('a')
+
+    def test_regex_lookup(self):
+        msg = 'subclasses of BaseDatabaseOperations may require a regex_lookup() method'
+        with self.assertRaisesMessage(NotImplementedError, msg):
+            self.base.regex_lookup(lookup_type='regex')
+
+    def test_set_time_zone_sql(self):
+        sql = self.base.set_time_zone_sql()
+        self.assertIsInstance(sql, str)
+        self.assertEqual(sql, '')
+
+    def test_sql_flush(self):
+        msg = 'subclasses of BaseDatabaseOperations must provide an sql_flush() method'
+        with self.assertRaisesMessage(NotImplementedError, msg):
+            self.base.sql_flush(None, None, None)
+
+    @skipIfDBFeature('supports_temporal_subtraction')
+    def test_subtract_temporals(self):
+        duration_field_internal_type = DurationField.get_internal_type()
+        msg = (
+            'This backend does not support %s subtraction.' %
+            duration_field_internal_type
+        )
+        with self.assertRaisesMessage(NotImplementedError, msg):
```

Comment:
According to https://www.python.org/dev/peps/pep-0249/ this should actually be `NotSupportedError`. Can I change it in the same PR? The same applies to `NotImplementedError` in `distinct_sql`.

Reply:
You can change it in a separate commit in the same PR.

Reply:
I made the change the first commit in this PR. I'm not sure about the commit message (mainly the header).